### PR TITLE
Add buzzer.cpp and support for U8Glib to Makefile

### PR DESCRIPTION
**Pull request by thawkins**
Adds support for compiling  Configurations with U8Glib graphics library to Makefile, Set the variable U8GLIB to 1 

Adds buzzer.cpp to list of target files so lcd support works.

### CODE REVIEW

**Comment by AnHardt** [comment]
Any other `Makers`, with an opinion, out there.

**Comment by thawkins** [comment]
Im probaly the only person who uses the "Makefile"

**Comment by Wackerbarth** [comment]
And just why do you do so?  Also, on what OS do you run? Do you run from the checkout of a git repository, or is it from an archive file of the source?

**Comment by thawkins** [comment]
Linux, i use a c++ IDE (cLion) to edit the code and run the makefile to
build and upload the code. I have a tool object defined in cLion so i can
just rightclick the project and say "build" or "build and upload".

The arduino ide is too simple and I cant use my crossref and search tools.

**Comment by thawkins** [comment]
Re: git, i have a fork on github which i have cloned to my local machine,
i maintain a seperate working repo that is a copy of that. I can hack away
at my local version and use diff to workout what in need to push into my
local clone. From there i do pushes to my gork on github, and issue pull
requests from there.

If i use the clone of the github fork directly then it complicates managing
upstream merges and pushes, with simple things i have changed that are only
relevant to my printer and not the Marlin in general getting pushed
through.

I may ditch the second copy and just work on a private branch of the clone,
which will make merging and diffjng easier.

**Comment by Wackerbarth** [comment]
I do something similar, but use the Arduino IDE with an external editor.  As we do in our TravisCI setup, you might consider using the "arduino" command line tool to do your builds. That would avoid the issue of keeping the Makefile in sync with our builds.

Either way, I encourage you to activate the automatic versioning script (generate_version_header_for_marlin) as a precondition of your build.

As for git repositories, I haven't found any advantage in having more than my one local copy, a subset of which I push to my personal github clone of the MarlinFirmware/Marlin.

For each of my printers, I have a few personal customizations in a branch that I keep rebasing to "Development".

**Comment by AnHardt** [comment]
```
-ARDUINO_VERSION      ?= 105
+ARDUINO_VERSION      ?= 106
```

@Wackerbarth 
The command line comes with 1.6.?

**Comment by Wackerbarth** [comment]
Yes, the command line (a useful version) is in 1.6. I think that 1.5 had a version which was less than adequate.

I don't see any good reason to use the 1.0 IDE. If you really need the cores or tools from that version, you can port them over to 1.6 with little effort and even run them in parallel.

The only issue that I have with IDE 1.6.6 is that it should be called 1.7 because they have, again, changed the file structure.

In the same vein, I cannot accept our current "Development" as 1.0.3 since the configuration files are not compatible with 1.0.2.

**Comment by Nandox7** [comment]
@thawkins I do only use the Makefile.
Or well, I used to because after some of the last changes in the Arduino IDE it go harder to get the propers paths.
I really hate that junky Arduino IDE so if the Makefile is kept would be perfect.

I may give it another go to make it work with 1.6.6.

**Comment by thawkins** [comment]
Im using 1.0.6  not 1.6.x

Most linux distros have 1.0.6 in thier repos.  I can setup a copy of 1.6.x
and see if it still works ok.

File: Marlin/Makefile [hunk 1]
```diff
@@ -37,12 +37,14 @@
 # from the commandline with "make HARDWARE_MOTHERBOARD=71" for example
 
 # This defined the board you are compiling for (see boards.h for the options)
-HARDWARE_MOTHERBOARD ?= 11
+HARDWARE_MOTHERBOARD ?= 33
 
 # Arduino source install directory, and version number
 # On most linuxes this will be /usr/share/arduino
+# On Mac OsX its probaly /Applications/Arduino.app/Contents/Resources/Java
```
Comment (thinkyhead) [inline Marlin/Makefile:44]: Spelling correction:

```
# On OS X it's probably /Applications/Arduino.app/Contents/Resources/Java
```

**Comment by sorki** [comment]
u8glib addition - #2478
buzzer.cpp addition -  #2479

The rest of the commits (RIGIDBOT related) should be extracted from this PR and moved to new one.

**Comment by thawkins** [comment]
There are no references to RIGID BOT in the PR, I'm not sure where people got that from...

**Comment by sorki** [comment]
You have to rebase your PR so it doesn't contain 47 commits which is confusing.

**Pull request closed without merging**